main.cpp

```diff
@@ -8,49 +8,58 @@
 #endif
 
 wgpu::Instance instance;
+wgpu::Adapter adapter;
 wgpu::Device device;
 wgpu::RenderPipeline pipeline;
 
-wgpu::SwapChain swapChain;
+wgpu::Surface surface;
 const uint32_t kWidth = 512;
 const uint32_t kHeight = 512;
 
-void SetupSwapChain(wgpu::Surface surface) {
-  wgpu::SwapChainDescriptor scDesc{
-      .usage = wgpu::TextureUsage::RenderAttachment,
-      .format = wgpu::TextureFormat::BGRA8Unorm,
+void ConfigureSurface() {
+  wgpu::SurfaceCapabilities capabilities;
+  surface.GetCapabilities(adapter, &capabilities);
+
+  wgpu::SurfaceConfiguration config{
+      .device = device,
       .width = kWidth,
       .height = kHeight,
-      .presentMode = wgpu::PresentMode::Fifo};
-  swapChain = device.CreateSwapChain(surface, &scDesc);
+      .format = capabilities.formats[0],
```

Comment:
Use the preferred format instead of format[0], I think, though I have filed https://github.com/webgpu-native/webgpu-headers/issues/290 about the fact that this is not clear.

Reply:
I'll be happy to replace this line with `.format = surface.GetPreferredFormat(adapter),` but I wonder if https://github.com/webgpu-native/webgpu-headers/issues/290 will get resolved soon and maybe we'll end up using `formats[0]`.